src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,250 @@
+/**
+ * @file methods/kernel_svm/kernel_svm_impl.hpp
+ * @author Himanshu Pathak
+ *
+ * Implementation of Kernel SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+#define MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "kernel_svm.hpp"
+
+namespace mlpack {
+namespace svm {
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t numClasses,
+    const double delta,
+    const double C,
+    const std::string& kernelFunction,
+    const bool fitIntercept,
+    const size_t max_iter,
+    const double tol) :
+    numClasses(numClasses),
+    delta(delta),
+    C(C),
+    kernelFunction(kernelFunction),
+    fitIntercept(fitIntercept),
+    max_iter(max_iter)
+{
+  Train(data, labels, numClasses, max_iter, tol);
+}
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const size_t inputSize,
+    const size_t numClasses,
+    const double delta,
+    const double C,
+    const std::string& kernelFunction,
+    const bool fitIntercept) :
+    inputSize(inputSize),
+    numClasses(numClasses),
+    delta(delta),
+    C(C),
+    kernelFunction(kernelFunction),
+    fitIntercept(fitIntercept)
+{
+  // No training to do here.
+}
+
+template<typename MatType, typename KernelType>
+double KernelSVM<MatType, KernelType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t numClasses,
+    const size_t max_iter,
+    const double tol)
+{
+  alpha = arma::zeros(data.n_cols);
+  size_t count = 0;
+  b = 0;
+  arma::vec E = arma::zeros(data.n_cols, 1);
+  double eta = 0;
+  double L = 0;
+  double H = 0;
+  arma::mat K;
+
+  if (kernelFunction == "linear")
+  {
+    K = data.t() * data;
+  }
+
+  else if (kernelFunction == "gaussian")
+  {
+    arma::vec X2 = arma::sum(arma::pow(data, 2), 2);
+    K = X2 + (X2.t() - 2 * data * data.t());
+  }
+  else
+  {
+    K = arma::zeros(data.n_rows, data.n_cols);
+  }
```

Comment:
If we use `KernelType` see comment above, we can probably replace the lines with `kernel.Evaluate(data)`, and don't have to test the kernel code again.

Reply:
Yup, will change it once I got my code working.

Reply:
Hey, can you make these changes first.